proto/tendermint/types/types.proto

```diff
@@ -219,6 +219,29 @@ message TxProof {
   repeated NMTProof proofs    = 3;
 }
 
+// SharesProof represents an NMT proof for a set of shares to the data root.
+message SharesProof {
+  // data raw shares that are proven.
+  repeated bytes data = 1;
+  // shares_proof NMT proofs for the raw shares. Could be more than one
+  // if the shares span more than one row.
+  repeated NMTProof shares_proof = 2;
+  // namespace_id namespaceID of the shares.
```

Comment:
```suggestion
  // namespace_id is the namespaceID of the shares.
```

Reply:
[question] is there a requirement that shares in this inclusion proof only belong to one namespace ID? Where is such a requirement enforced?

Reply:
it is an implementation limitation. creating inclusion proofs for multiple namespaces:

- more complex
- would require making changes to the `nmt` repository. Since it only supports proofs for a single namespace.

Also, from a QGB perspective, we will want to prove that some data was posted to a certain namespace. 

Ideally, it would be awesome to support any shares without constraints. But, it would take more time/effort to have it  working.